### PR TITLE
Don't use ImageGrid in demo_text_rotation_mode.

### PR DESCRIPTION
**Pull request by anntzer**
The example can be written without involving axes_grid.

old: https://matplotlib.org/gallery/text_labels_and_annotations/demo_text_rotation_mode.html#sphx-glr-gallery-text-labels-and-annotations-demo-text-rotation-mode-py
new:
new
(admittedly the inability to set the absolute spacing between subplots is mildly annoying)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/text_labels_and_annotations/demo_text_rotation_mode.py [hunk 1]
```diff
@@ -17,62 +17,69 @@
...
 
 def test_rotation_mode(fig, mode, subplot_location):
     ha_list = ["left", "center", "right"]
     va_list = ["top", "center", "baseline", "bottom"]
-    grid = ImageGrid(fig, subplot_location,
-                     nrows_ncols=(len(va_list), len(ha_list)),
-                     share_all=True, aspect=True, cbar_mode=None)
+    axs = np.empty((len(va_list), len(ha_list)), object)
+    gs = subplot_location.subgridspec(*axs.shape)
+    axs[0, 0] = fig.add_subplot(gs[0, 0])
```
Comment (anntzer) [inline examples/text_labels_and_annotations/demo_text_rotation_mode.py]: this part would be slightly easier with something like https://github.com/matplotlib/matplotlib/pull/14421 available.

**Comment by ImportanceOfBeingErnest** [comment]
I don't like that the orange boxes now touch the vertical spines.  
The spacing between axes could be made smaller. 

In general the old example was very good (it got improved a lot not too long ago already). If it is more complicated to not use `ImageGrid`, I think one could just stay with it.

**Comment by anntzer** [comment]
hspace=wspace=0 gives
new

I don't mind waiting for the discussion on #14421 to be resolved first.

**Comment by timhoffm** [comment]
What's the motivation for the change? I don't see that either one is particularly be better than the other.

**Comment by anntzer** [comment]
I *think* gridspecs are nicer to use -- even though ImageGrid does allow for more accurate positioning, you need to use mpl_toolkits-specific APIs for e.g. labels and ticks (https://github.com/matplotlib/matplotlib/pull/15281/files#diff-90eb9024490f32de5d0796d7456283f0L35, https://github.com/matplotlib/matplotlib/pull/15281/files#diff-90eb9024490f32de5d0796d7456283f0L51) which is a bit awkward.

**Comment by jklymak** [comment]
I agree w/ @anntzer.  I think, when possible, we should make examples use the core library, not things from `mpl_toolkits`.  If thats not possible, I think we should work on making it possible.

**Comment by jklymak** [comment]
If you rebase, please ping me for a review.  I think we should change this.

**Comment by anntzer** [comment]
rebased

**Comment by timhoffm** [comment]
Not a prerequisite for merging, but would #20016 help with the spacing?

**Review by timhoffm — APPROVED** [review]

**Comment by anntzer** [comment]
possibly?

**Pull request merged**